site_scons/site_tools/compiler_setup.py

```diff
@@ -55,6 +55,9 @@ def _base_setup(env):
     env.Append(CCFLAGS=['-g', '-Wextra', '-Wshadow', '-Wall', '-fpic'])
 
     env.AppendIfSupported(CCFLAGS=DESIRED_FLAGS)
+    env.AppendIfSupported(LINKFLAGS=DESIRED_FLAGS)
+    if '-fstack-protector-strong' in env["CCFLAGS"]:
+        env.AppendENVPath("CGO_LDFLAGS", "-fstack-protector-strong", sep=" ")
```

Comment:
It looks like `_append_if_supported` always appends for LINKFLAGS, so this is misleading